File: Kevin van den Bekerom/DSLQL/src/main/java/org/uva/sea/ql/ast/expr/Pos.java
package org.uva.sea.ql.ast.expr;

import org.uva.sea.ql.ast.form.Context;
import org.uva.sea.ql.ast.visit.Visitor;
import org.uva.sea.ql.type.NumericalType;
import org.uva.sea.ql.type.Type;

public class Pos extends UnaryExpr {
	
	public Pos(Expr expr) {
		super.child = expr;
	}
	
	@Override
	public Integer eval() {
		return Math.abs( (Integer) child.eval());
	}
	
	public void accept(Visitor visitor, Object context) {
		visitor.visit(this, context);
	}

	@Override
	public String toString() {
		return super.toString() + " (positive)";
	}
	
	@Override
	public Type getType(Context context) {
		return new NumericalType();
	}
}

File: Edward Poot/src/ql/ast/expr/unaryoperatorexpr/pos.go
package unaryoperatorexpr

import (
	"math"
	"ql/ast/expr"
)

type Pos struct {
	UnaryOperator
}

func NewPos(value expr.Expr) Pos {
	return Pos{UnaryOperator{Value: value}}
}

func (p Pos) Eval(s interface{}) interface{} {
	return int(math.Abs(float64(p.Value.Eval(s).(int))))
}
